pi-top/pi-topd
# Support legacy devices and accessories

| Status  | Ticket/Issue |
| :---: | :--: |
| Ready| [Ticket](https://pi-top.atlassian.net/browse/OS-1222) |


#### Main changes

This PR updates pi-topd to support legacy devices and accessories. 

Tested this with:
- pi-top[3] with v2 speaker, pi-topPULSE and pi-topPROTO+
- pi-topCEED with v1 & v2 speaker, pi-topPULSE and pi-topPROTO+
- pi-top[4] with pi-topPULSE
- Original pi-top with v1 speaker

#### Screenshots (feature, test output, profiling, dev tools etc)
N/A

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
N/A

#### Manual testing tips
You'll need an already onboarded OS to test this since the web-portal will fail on legacy devices due to the error that this PR fixes.
- Download the .debs and install them in your pi-top.
- Make sure the `pi-topd` service works correctly: `sudo systemctl restart pi-topd` and then  `systemctl status pi-topd`  should report that the service is active and running.
- Try restarting the web-portal: `sudo systemctl restart pt-os-web-portal` and `systemctl status pt-os-web-portal`  should report that the service is active and running.
- If you have any pi-top accessories, you can plug them into your legacy device and test it.

Another way to test if the onboarding works for legacy devices using this PR artifacts would be:
- Download the artifacts into your laptop.
- Burn a bullseye OS into a SD card.
- Copy the artifacts to the boot folder in the SD card (eg: `cp *.deb /Volumes/boot/`)
- Enable ssh on boot (`touch /Volumes/boot/ssh`)
- Remove the SD card and plug it into a pi-top[4]
- Turn on your pi-top[4]. When it's ready, ssh into it
- Install the debs with `sudo apt install /boot/*.deb`
- Turn off your pi-top[4] and remove the SD card.
- Now try using that SD card to boot a pi-top[3] or a pi-topCEED. Onboarding should come up on screen.


#### Tag anyone who definitely needs to review or help
N/A
